forms/AbiEncodeForm.vue

```diff
@@ -0,0 +1,224 @@
+        :icon-left="$icons.plus"
+        @click="addArg"
+      />
+    </div>
+    <div class="abi-encode-form__output">
+      <h2>{{ $t('abi-encode-form.output-title') }}</h2>
+      <input-field
+        :model-value="funcSignature"
+        :label="$t('abi-encode-form.func-signature-label')"
+        :placeholder="$t('abi-encode-form.func-signature-placeholder')"
+        readonly
+      />
+      <textarea-field
+        :model-value="abiEncoding"
+        :label="$t('abi-encode-form.abi-encoding-label')"
+        :placeholder="$t('abi-encode-form.abi-encoding-placeholder')"
+        readonly
+      />
+      <app-button
+        :text="$t('abi-encode-form.abi-encoding-copy-btn')"
+        @click="() => copyToClipboard(abiEncoding)"
+      />
+    </div>
+  </form>
+</template>
+
+<script lang="ts" setup>
+import { AppButton } from '#components'
+import { useFormValidation } from '@/composables'
+import { ETHEREUM_TYPES } from '@/enums'
+import { InputField, SelectField, TextareaField } from '@/fields'
+import {
+  ErrorHandler,
+  copyToClipboard,
+  createFuncArgTypeRule,
+  createFuncArgValueRule,
+  forEach,
+  parseFuncArgToValueOfEncode,
+} from '@/helpers'
+import { type AbiEncodeForm } from '@/types'
+import { Interface } from 'ethers'
+import { v4 as uuidv4 } from 'uuid'
+import { computed, reactive, ref, watch } from 'vue'
+
+defineProps<{
+  title: string
+}>()
+
+const abiEncoding = ref('')
+const funcSignature = ref('')
+
+const resetOutput = () => {
+  abiEncoding.value = ''
+  funcSignature.value = ''
+}
+
+const form = reactive({
+  funcName: '',
+  args: [] as AbiEncodeForm.FuncArg[],
+})
+const rules = computed(() => ({
+  args: {
+    $each: forEach({
+      type: { funcArgTypeRule: createFuncArgTypeRule() },
+      value: { funcArgValueRule: createFuncArgValueRule() },
+    }),
+  },
+}))
+
+const { getFieldErrorMessage, isFieldsValid, touchField } = useFormValidation(
+  form,
+  rules,
+)
+
+const addArg = () => form.args.push({ id: uuidv4(), type: '', value: '' })
+const removeArg = (id: AbiEncodeForm.FuncArg['id']) => {
+  form.args = form.args.filter(arg => arg.id !== id)
+}
+
+const funcArgErrorMsgLists = computed<string[][]>(
+  () => getFieldErrorMessage('args') as unknown as string[][],
+)
+
+const getFuncArgErrorMsg = (
+  id: AbiEncodeForm.FuncArgErrorMsgInfo['id'],
+  field: AbiEncodeForm.FuncArgErrorMsgInfo['field'],
+): AbiEncodeForm.FuncArgErrorMsgInfo['message'] => {
+  let funcArgErrorMsgInfo: AbiEncodeForm.FuncArgErrorMsgInfo | null = null
+
+  for (const list of funcArgErrorMsgLists.value) {
+    const msg = list.find(msg => {
+      try {
+        const msgInfo = JSON.parse(msg) as AbiEncodeForm.FuncArgErrorMsgInfo
+        return msgInfo.id === id && msgInfo.field === field
+      } catch {
+        return false
+      }
+    })
+
+    if (msg) {
+      funcArgErrorMsgInfo = JSON.parse(msg)
+      break
+    }
+  }
+
+  return funcArgErrorMsgInfo?.message ?? ''
```

Comment:
will only return '' when funcArgErrorMsgInfo.message = null, are you sure about that?

Reply:
discussed, replaced by ||